rasa/core/policies/ensemble.py

```diff
@@ -562,6 +563,7 @@ def _pick_best_policy(
             best_prediction.policy_priority,
             policy_events,
             is_end_to_end_prediction=best_prediction.is_end_to_end_prediction,
+            entities=best_prediction.entities,
```

Comment:
there is no need to add entities to `PolicyPrediction`, you can add `DefinePrevUserUtteredEntities` directly in TED as optional event. Similar like it is done in `RulePolicy` with `LoopInterrupted` except that `RulePolicy` adds it as `events` (applicable regardless of which policy wins), while TED should add it as `optional_events` (applicable only if TED wins)

Reply:
ah, thanks! will update the code